python/docs/source/conf.py

```diff
@@ -102,9 +102,9 @@
 .. |examples| replace:: Examples
 .. _examples: https://github.com/apache/spark/tree/{0}/examples/src/main/python
 .. |downloading| replace:: Downloading
-.. _downloading: https://spark.apache.org/docs/{1}/building-spark.html
+.. _downloading: https://spark.apache.org/docs/{1}/#downloading
 .. |building_spark| replace:: Building Spark
-.. _building_spark: https://spark.apache.org/docs/{1}/#downloading
+.. _building_spark: https://spark.apache.org/docs/{1}/building-spark.html
```

Comment:
If this happens in Apache Spark 3.5.0, could you add `3.5.0` to the affected version, @panbingkun ?

<img width="302" alt="Screenshot 2023-11-15 at 3 07 23 PM" src="https://github.com/apache/spark/assets/9700541/d3966d02-8572-4a96-b56a-a4bf729e65f9">

Reply:
Yes, I have checked the branches: branch-3.3, branch-3.4, and branch-3.5, which have all been affected. Therefore, I have added: 3.5.0, 3.4.1, 3.3.3. If there are conflicts during the merge process, please let me know and I will resubmit them on each branch. Thank you very much for your reminder.
<img width="624" alt="image" src="https://github.com/apache/spark/assets/15246973/00795396-3aef-47af-ab39-daff66686228">